language/move-vm/types/src/values/values_impl.rs

```diff
@@ -2178,57 +2184,136 @@ impl Struct {
  *   transaction execution the dirty ones can be identified and wrote back to storage.
  *
  **************************************************************************************/
-impl GlobalValue {
-    pub fn new(v: Value) -> PartialVMResult<Self> {
-        match v.0 {
-            ValueImpl::Container(Container::StructR(r)) => {
-                // TODO: check strong count?
-                Ok(Self {
-                    status: Rc::new(RefCell::new(GlobalDataStatus::Clean)),
-                    fields: r,
-                })
+impl GlobalValueImpl {
+    fn cached(val: ValueImpl, status: GlobalDataStatus) -> PartialVMResult<Self> {
```

Comment:
The only caller of this passes in `status: GlobalDataStatus::Clean`. Maybe we should simplify this API accordingly (+ to match the old code)?

Reply:
`move_to` calls this with `Dirty` on line 2242.